### PR TITLE
Force-allow to set paths

### PR DESCRIPTION
**Pull request by Redsandro**
Set paths even if some variable is blocking the namespace (AKA path) using an optional 4th parameter `force`

### CODE REVIEW

File: dottie.js [hunk 1]
```diff
@@ -81,7 +81,13 @@
...
       }
 
-      if (typeof current !== 'object') throw new Error('Target key is not suitable for a nested value (it is either not undefined or not an object)');
+      if (typeof current !== 'object') {
+        // If force === true, bruteforce the path without throwing errors.
+        if (force === true)
+          current = {};
+        else
+          throw new Error('Target key is not suitable for a nested value (it is either not undefined or not an object)');
+      }
```
Comment (mickhansen) [inline dottie.js]: Please use brackets for if statements

```
if {

} else {

}
```

**Comment by mickhansen** [comment]
Please include a test for this behaviour.

**Comment by Redsandro** [comment]
Hmm.. on second thought, why does 

```
      if (!hasOwnProp.call(current, piece) || current[piece] === undefined) {
        current[piece] = {};
      }
```

come before

```
      if (typeof current !== 'object') throw new Error
```

I'm not sure my code works the way I thought

**Comment by mickhansen** [comment]
Yeah i'm not exactly sure either, been a while since i wrote that code.

**Comment by Redsandro** [comment]
I'm planning to figure out how to create tests at some point. On hold. :P

**Comment by mickhansen** [comment]
Published in `v1.0.0`

**Comment by Redsandro** [comment]
Making tests for this was still on my ever-growing TO DO list.. but nice you published this already :smile:

**Comment by mickhansen** [comment]
I added some tests myself and converted it to an option instead, so `.set(obj, key, value, {force: true})`

**Comment by Redsandro** [comment]
Thank you. :+1:

**Pull request merged**